### ISSUE TITLE
Python 2.7 autcompletion not working

### ISSUE THREAD

**Issue by SirJls** [proofit404/anaconda-mode#329]
I'm currently trying to get working auto-completion for python 2.7 using anaconda-mode on windows 10. The endeavor so far is not very fruitful... I'm currently using spacemacs  with the following settings:

#### System Info :computer:
- OS: windows-nt
- Emacs: 26.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 1c166c2)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm auto-completion emacs-lisp neotree python)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS THREADS LCMS2

### Problem description🐞

When firing up emacs, opening `*.py` file and activating a virtual python 2.7 environment the `*anaconda-mode*` buffer tells me one of the modules is trying to call a python 3 functions, namely clear on a list object... 

The error is as follows:

```
Traceback (most recent call last):
  File "<string>", line 61, in <module>
AttributeError: 'list' object has no attribute 'clear'

```

The cause of this problem is that `anaconda-mode` is automatically acquiring the dependencies for python 3.7, which is what I don't want... Can I disable this behavior and use pip to install the dependencies manually? Or is it possible to instruct `anaconda-mode` to get the 2.7 dependencies? Any help would be most welcome, thank you!

**Comment by thinkiny** [comment]
same problem

**Comment by thinkiny** [comment]
change anaconda-mode.el:98 missing_dependencies = [] works

**Comment by thinkiny** [comment]
change anaconda-mode.el:98 missing_dependencies = [] works, remember delete anaconda-mode.elc

**Comment by SirJls** [comment]
Okay, I took some time walking through the code and learned that the problem lies in the missing_dependencies list as mentioned by @thinkiny and not in the versioning of the acquired  dependencies by anaconda-mode. This came to my attention as I have no problem running the code with a version of Python>=3.3

The fix is really issue as well: anaconda-mode clears the list by using a Python>=3.3 attribute on list objects, namely `clear()`. This can easily be solved by replacing it with `del missing_dependencies[:]` ensuring backwards compatibility with Python2.7 as the support for this version is currently pushed back to 2020 (source: PEP 373).

As a side note, I found another strange issue related to jedi while fixing this problem. When using a virtual environment, jedi searches for a `python.exe` in the `Scripts` directory. I'm currently using anaconda2 and creating a virtual environment using `conda create --name <myenvname> python=<your version> <list of packages>` install the python environment in the parent folder of the `Scripts` directory. This could however easily be fixed by using a symlink, but using a company laptop I don't have the privileges to execute this command?! To fix the issue I have to copy the whole executable and accompanying libraries and a better solution for this problem would be nice.
